### PR TITLE
Create new service,dto and mapper

### PR DESCRIPTION
**Pull request by qetiavanashvili**
1.I have created  DogBreedsListDTO.
2.I have created DogBreedService file.
3. I have implemented mapper: DogBreedMapper

https://github.com/ashalva/ios.memenator/issues/2

### CODE REVIEW

File: GeorgianMemenator/Network/Services/DogBreedService.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        
+        return try! request
+            .usingMethod(.GET)
+            .build()
+            .asFuture()
+        
+            .map {
+                $0.data.map { breed in DogBreedsMapper.toEntity(dto: breed) }
+            }
+//            .compactMap({ DogBreedMapper.toEntity(dto: $0) })
```
Comment (ashalva) [inline GeorgianMemenator/Network/Services/DogBreedService.swift]: Unneeded comment, please delete

File: GeorgianMemenator/Network/Mappers/DogBreedsMapper.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        .init(
+            name: dto.attributes.name,
+            description: dto.attributes.description,
+            minLife: Int(dto.attributes.life.min),
+            maxLife: Int(dto.attributes.life.max),
+            maleMinWeight: Int(dto.attributes.maleWeight.min),
+            maleMaxWeight: Int(dto.attributes.maleWeight.max),
+            femaleMinWeight: Int(dto.attributes.femaleWeight.min),
+            femaleMaxWeight: Int(dto.attributes.femaleWeight.max)
+
```
Comment (ashalva) [inline GeorgianMemenator/Network/Mappers/DogBreedsMapper.swift]: Unneeded empty line, please try to delete them, I have been mentioning it for long time already.

File: GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
+//
+//  DogBreedsListDTO .swift
+//  GeorgianMemenator
+//
+//  Created by Qeti Avanashvili on 25.02.23.
+//
+
+import Foundation
+struct DogBreedDTO: Codable {
```
Comment (ashalva) [inline GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift:9]: There should be empty line after import and struct

File: GeorgianMemenator/Network/BusinesssModels/DogBreed.swift [hunk 4]
```diff
@@ -7,10 +7,17 @@
...
-    let maxAge: Int
+    let id: String = UUID().uuidString
+    let description: String
+    let minLife: Int
+    let maxLife: Int
+    let maleMinWeight: Int
+    let maleMaxWeight: Int
+    let femaleMinWeight: Int
+    let femaleMaxWeight: Int
+
```
Comment (ashalva) [inline GeorgianMemenator/Network/BusinesssModels/DogBreed.swift]: Unneeded lines empty lines

File: GeorgianMemenator/Network/BusinesssModels/DogBreed.swift [hunk 5]
```diff
@@ -7,10 +7,17 @@
 //
 
 import Foundation
-
 struct DogBreed: Identifiable {
-    let id: String = UUID().uuidString
     let name: String
-    let minAge: Int
-    let maxAge: Int
+    let id: String = UUID().uuidString
```
Comment (ashalva) [inline GeorgianMemenator/Network/BusinesssModels/DogBreed.swift:12]: Can you move `id` as the first property? and then name etc ...

File: GeorgianMemenator/Network/Mappers/DogBreedsMapper.swift [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+//
+
+import Foundation
+
+enum DogBreedsMapper {
+    static func toEntity(dto: DogBreedDTO) -> DogBreed {
+        .init(
+            name: dto.attributes.name,
+            description: dto.attributes.description,
+            minLife: Int(dto.attributes.life.min),
```
Comment (ashalva) [inline GeorgianMemenator/Network/Mappers/DogBreedsMapper.swift]: There is no need to cast integers to Int, so you can safely remove `Int(dto.att...)`, just pass the `dto.attributes.life.min`. Its for all properties where you are casting them to Int.

File: GeorgianMemenator/Network/Services/DogBreedService.swift [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    func getDogBreeds() -> AnyPublisher<[DogBreed], Error> {
+        let request = Endpoint<DogBreedsListDTO>(baseURL: dogBreedApiUrl)
+            .appendingPathParameter("breeds")
+            .usingDefaultParameters()
+        
+        return try! request
+            .usingMethod(.GET)
+            .build()
+            .asFuture()
+
```
Comment (ashalva) [inline GeorgianMemenator/Network/Services/DogBreedService.swift]: Unneeded new line

File: GeorgianMemenator/Network/Services/DogBreedService.swift [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        
+        return try! request
+            .usingMethod(.GET)
+            .build()
+            .asFuture()
+            .map {
+                $0.data.map { breed in DogBreedsMapper.toEntity(dto: breed) }
+            }
+            .eraseToAnyPublisher()
+
```
Comment (ashalva) [inline GeorgianMemenator/Network/Services/DogBreedService.swift]: Empty line :|

File: GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    struct Attributes: Codable {
+        struct Weight: Codable {
+            let max, min: Int
+        }
+
+        struct Life: Codable {
+            let max, min: Int
+        }
+
+
```
Comment (ashalva) [inline GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift]: Empty line :|

File: GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift (hunk 3 above)
Comment (ashalva) [reply GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift:9]: You have not added new line here?

File: GeorgianMemenator/Network/Services/DogBreedService.swift (hunk 8 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Network/Services/DogBreedService.swift]: fixed

File: GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift (hunk 9 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift]: fixed

**Comment by qetiavanashvili** [comment]
I added the line after import

**Comment by qetiavanashvili** [comment]
I think, I fixed everything

File: GeorgianMemenator/Network/BusinesssModels/DogBreed.swift [hunk 10]
```diff
@@ -7,10 +7,14 @@
 //
 
 import Foundation
-
```
Comment (ashalva) [inline GeorgianMemenator/Network/BusinesssModels/DogBreed.swift]: We need new line between import and struct definition

**Review by ashalva — COMMENTED** [review]
All good, one small bit left and we can merge after that 👍

File: GeorgianMemenator/Network/BusinesssModels/DogBreed.swift (hunk 10 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Network/BusinesssModels/DogBreed.swift]: I have added, but you could not see as it seems, I dont know why

Comment (qetiavanashvili) [reply GeorgianMemenator/Network/BusinesssModels/DogBreed.swift]: Screenshot 2023-02-26 at 15 46 27

Comment (ashalva) [reply GeorgianMemenator/Network/BusinesssModels/DogBreed.swift]: @qetiavanashvili  you attached the screenshot for `DogBreedDTO`, but I am talking about `DogBreed`

**Comment by qetiavanashvili** [comment]
Okey, I see. I fixed

**Review by ashalva — APPROVED** [review]
Looks good 💹

**Pull request merged**